compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -352,14 +353,22 @@ fn mir_promoted(
 
     // What we need to run borrowck etc.
     let promote_pass = promote_consts::PromoteTemps::default();
+    let promote_array = promote_consts_local_arrays::PromoteArraysOpt::default();
     pm::run_passes(
         tcx,
         &mut body,
-        &[&promote_pass, &simplify::SimplifyCfg::PromoteConsts, &coverage::InstrumentCoverage],
+        &[
+            &promote_pass,
+            &promote_array,
```

Comment:
since this is something that should not have user-visible effects (e.g. affecting dropck, const eval UB or borrowck), it should be run as part of the regular runtime optimization pipeline